test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -43,6 +49,25 @@
               (println "timed-out waiting for events")
               @dest)))))))
 
+(defn wait-for-exactly-event
+  "Waits for event to land in dest. On first registered event or events in dest,
+  determines whether the registered event(s) match the supplied expected
+  event. If so, returns the expected event. Otherwise returns contents of dest."
+  [dest expected-event]
+  (let [start-time (System/currentTimeMillis)]
+    (loop []
```

Comment:
It looks like this could be a `while` instead of a `loop`/`recur` as there are no bindings for the loop.